report/elements/bench/cpu_frequency.py

```diff
@@ -0,0 +1,88 @@
+from elements.bench.base import AbstractBench
+
+import matplotlib.pyplot as plt
+import numpy as np
+import os.path
+from matplotlib.ticker import MultipleLocator
+
+class CpuFrequency(AbstractBench):
+    def __init__(self, obj, bench_obj):
+        self.obj = obj
+        self.bench_obj = bench_obj
+
+    def to_html(self):
+        self.gen_images()
+        wd = os.getcwd()
+
+        header = "<h2 id='CPUFrequency'>CPU Frequency</h2>"
+        p1 = "<p> All values are in GHz. </p>"
+
+        imgs = f"<img src='{wd}/out/cpu_frequency_multiplot.png'/>"
+        
+        return header + p1 + imgs
+
+    def gen_images(self):
+        data = self.bench_obj["results"]
+
+        div = 0
+        for k in range(8, 1, -1) :
+            if len(data) > k and len(data)%k == 0 :
+                div = k
+                break
+
+        figure, truc = plt.subplots(div, (len(data)//div))
+        plt.subplots_adjust(hspace=1, wspace=0.3)
+
+        max_val = 0
+        min_val = 100
+
+        for cpu in data:
+            for vals in data[cpu]:
+                max_val = max(max_val, max(*vals))
+                min_val = min(min_val, min(*vals))
+
+        max_val *= 1.1
+        min_val *= 0.9
+
+        all_vals = []
+
+        for j, cpu in enumerate(data):
+            values = []
+            top = []
+            bot = []
+            if j == 0 :
+                mot = ' Core'
+            else :
+                mot = ' Cores'
+            for vals in data[cpu]:
+                temp = []
+                for val in vals:
+                    temp.append(val)  
+                values.append(np.mean(temp))
+                all_vals.append(values[-1])
+
+            if div != 1 :
+                cadre = truc[j//((len(data)//div)), j%((len(data)//div))] 
+            else :
+                cade = truc[j//((len(data)//div))]
+            cadre.plot(range(1, j + 2), values, marker=".")
```

Comment:
```suggestion
            cadre.plot(range(1, j + 2), values, marker=".", label=str(j+1) + mot)
```

And one more for label 😄